File: src/me/tcklpl/naturaldisaster/util/ActionBar.java
package me.tcklpl.naturaldisaster.util;

import me.tcklpl.naturaldisaster.reflection.Packets;
import me.tcklpl.naturaldisaster.reflection.ReflectionUtils;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class ActionBar {

    private Object packet;

    public ActionBar(String text) {
        this.packet = Packets.Play.PlayOutChat("{\"text\":\"" + text + "\"}", Packets.Play.ChatMessageType.GAME_INFO);
    }

    public void sendToPlayer(Player p) {
        ReflectionUtils.sendPacket(p, packet);
    }

    public void sendToAll() {
        for (Player p : Bukkit.getOnlinePlayers())
            sendToPlayer(p);
    }
}

File: src/main/java/me/tcklpl/naturaldisaster/util/ActionBar.kt
package me.tcklpl.naturaldisaster.util

import net.md_5.bungee.api.ChatMessageType
import net.md_5.bungee.api.chat.TextComponent
import org.bukkit.Bukkit
import org.bukkit.entity.Player

class ActionBar(text: String) {
    private val textComponent: TextComponent = TextComponent(text)

    fun sendToPlayer(p: Player) {
        p.spigot().sendMessage(ChatMessageType.ACTION_BAR, textComponent)
    }

    fun sendToAll() {
        for (p in Bukkit.getOnlinePlayers()) sendToPlayer(p)
    }
}
